lib/ansiblelint/rules/__init__.py

```diff
@@ -49,14 +53,14 @@ def create_matcherror(
             details: str = "",
             filename: str = None) -> MatchError:
         return MatchError(
-            message=message,
+            message=message or self.shortdesc,
```

Comment:
It is normal that you did not see any rule using custom message as it was not possible but I can give you a practical example were this is needed.

Remember about galaxy.yml file and the platforms that you can write there, some values may be invalid. That means that a single rule may want to generate something like "Platform CentOS was not found among the list recognized by galaxy server". If you list multiple ones, you may get multiple errors.

In fact the invalid role-name does have a variant error message that includes the invalid role name, but is not really affected by this limitation because we will never have a single role generating two invalid name errors.

Another example that used in the past was a hypothetical spell checker which could identify two different bad words on the same file and line.

The same kind of issue could appear on a line that looks like `foo: [invalid-value-a, invalid-value-b]`, where same rule may identify two separated issues: `invalid-value-a is not defined` and `invalid-value-b is not defined`. Keep in mind that some lines are quite long (aka shell/command can be entire scripts).

As these changes will only go into 5.0 version we can afford to break things in order to correct historical mistakes. Still, I plan to also minimize breaking changes and make them only were there are benefits.

Reply:
> Remember about galaxy.yml file and the platforms that you can write there, some values may be invalid. That means that a single rule may want to generate something like "Platform CentOS was not found among the list recognized by galaxy server". If you list multiple ones, you may get multiple errors.

I am pretty sure this is not a kind of error ` match` is expected to find since it is expected to operate on a single line, not a structure of some sort. I am not familiar enough with the codebase to be sure, but I would expect this to be handled by some other method that operates on more than a single line.

> Another example that used in the past was a hypothetical spell checker which could identify two different bad words on the same file and line.
>
> The same kind of issue could appear on a line that looks like `foo: [invalid-value-a, invalid-value-b]`, where same rule may identify two separated issues: `invalid-value-a is not defined` and `invalid-value-b is not defined`. Keep in mind that some lines are quite long (aka shell/command can be entire scripts).

These are all valid scenarios. So I guess we do need to be able to return multiple error messages. But maybe we can just return back a list of strings? Forcing each `match` method to mess around with line numbers feels like a bad design choice since the caller that iterates over the file already has this information. If we return back a list of strings, we probably get the best of both worlds: the match method can be oblivious to the line number it operates on while still being able to return multiple messages, and the match error construction stays centralized in one place that already has enough content to build it.

Reply:
Github highaboveus2